### PR TITLE
Reduce sprintf overhead in code showing up during sync profiles

### PR DESCRIPTION
**Pull request by ValarDragon**
Reduce some sprintf overheads

This is a pretty micro-optimization. It likely has more impact for txs than epoch. In epoch it shaves like 100ms (there are way better things to shave time on there)

### CODE REVIEW

File: x/poolmanager/types/keys.go [hunk 1]
```diff
@@ -43,7 +44,16 @@ var (
 
 // ModuleRouteToBytes serializes moduleRoute to bytes.
 func FormatModuleRouteKey(poolId uint64) []byte {
-	return []byte(fmt.Sprintf("%s%d", SwapModuleRouterPrefix, poolId))
+	// Estimate the length of the string representation of poolId
+	// 7 is a safe upper bound, (9999999) pools, and is an 8 byte allocation (not bad)
```
Comment (nicolaslara) [inline x/poolmanager/types/keys.go:48]: just for extra safety... should we check that poolId < 9999999?

Comment (ValarDragon) [reply x/poolmanager/types/keys.go:48]: All we could really do is panic, which is what will already happen here anyway

Comment (ValarDragon) [reply x/poolmanager/types/keys.go:48]: We can add that check to CreatePool

Comment (ValarDragon) [reply x/poolmanager/types/keys.go]: Ok just raised the upperbound to `99,999,999,999`. At $1 to create a pool, this is a third the market cap of ETH, so going to go ahead and merge!

**Comment by ValarDragon** [comment]
Added check for it in #7203

**Pull request merged**